crates/networking/p2p/sync.rs

```diff
@@ -480,15 +491,15 @@ impl SyncManager {
     }
 
     fn add_blocks(
-        &self,
+        blockchain: Arc<Blockchain>,
```

Comment:
Nit:
```suggestion
        self: Arc<Self>,
```
Otherwise, see the other comment, I think it allows dropping all changes to this function.

Reply:
See answer below.